compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp

```diff
@@ -404,10 +404,18 @@ struct GPUPadEncodingLayoutResolverAttrInterface final
     : Encoding::EncodingLayoutResolverAttrInterface::ExternalModel<
           GPUPadEncodingLayoutResolverAttrInterface, GPUPadLayoutAttr> {
   Attribute cloneWithSimplifiedConfig(Attribute attr,
-                                      DictionaryAttr /*config*/) const {
-    // This attribute is self-contained and does not need to look anything up
-    // from the target `config`.
-    return attr;
+                                      DictionaryAttr config) const {
+    MLIRContext *ctx = attr.getContext();
+    IREE::GPU::TargetAttr gpuTarget = getGPUTargetAttr(config);
+    if (!gpuTarget) {
+      return IREE::Encoding::IdentityEncodingAttr::get(ctx);
+    }
+    // GPUPadLayoutAttr is only enabled for CDNA2 and CDNA3 for the time being.
+    // TODO(kuhar): Enable for other HIP targets.
+    if (!llvm::is_contained({"gfx90a", "gfx942"}, gpuTarget.getArch())) {
+      return IREE::Encoding::IdentityEncodingAttr::get(ctx);
+    }
```

Comment:
I don't think we should move this code here -- I'd rather have it in KnownTargets together with all the other arch-specific information. We could materialize it in the target chip attribute if necessary, but I don't think nothing else needs it